src/main/java/org/sagebionetworks/bridge/reporter/worker/BridgeReporterProcessor.java

```diff
@@ -1,82 +1,62 @@
 package org.sagebionetworks.bridge.reporter.worker;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.JsonNodeFactory;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.google.common.base.Stopwatch;
 import org.joda.time.DateTime;
 import org.joda.time.LocalDate;
+import org.joda.time.MutableDateTime;
 import org.sagebionetworks.bridge.json.DefaultObjectMapper;
 import org.sagebionetworks.bridge.reporter.helper.BridgeHelper;
+import org.sagebionetworks.bridge.reporter.request.ReportScheduleName;
 import org.sagebionetworks.bridge.sdk.models.ResourceList;
 import org.sagebionetworks.bridge.sdk.models.reports.ReportData;
 import org.sagebionetworks.bridge.sdk.models.studies.StudySummary;
 import org.sagebionetworks.bridge.sdk.models.upload.Upload;
-import org.sagebionetworks.bridge.sqs.PollSqsCallback;
 import org.sagebionetworks.bridge.sqs.PollSqsWorkerBadRequestException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.counting;
 
-
 /**
  * SQS callback. Called by the PollSqsWorker. This handles a reporting request.
  */
 @Component
-public class BridgeReporterSqsCallback implements PollSqsCallback {
-    private static final Logger LOG = LoggerFactory.getLogger(BridgeReporterSqsCallback.class);
+public class BridgeReporterProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(BridgeReporterProcessor.class);
 
     private BridgeHelper bridgeHelper;
 
-    public enum REPORT_SUFFIX {
-        DAILY_SUFFIX("-daily-upload-report"),
-        WEEKLY_SUFFIX("-weekly-upload-report");
-
-        private final String suffix;
-
-        REPORT_SUFFIX(String s) {
-            this.suffix = s;
-        }
-
-        public String getSuffix() {
-            return this.suffix;
-        }
-    }
-
     @Autowired
     public final void setBridgeHelper(BridgeHelper bridgeHelper) {
         this.bridgeHelper = bridgeHelper;
     }
 
-    /** Parses the SQS message. */
-    @Override
-    public void callback(String messageBody) throws IOException, PollSqsWorkerBadRequestException {
+    /** Process the passed sqs msg as JsonNode. */
+    public void process(JsonNode body) throws Exception{
```

Comment:
It's generally considered bad form to throw a generic Exception in production code. This makes it so that callers don't know which exceptions to catch and will have to catch (or throw) all of them. It's better to list the specific exceptions you throw.